objects/obj_pause/Draw_77.gml

```diff
@@ -0,0 +1,47 @@
+//Disable alpha blending (Drawing just RGB)
+gpu_set_blendenable(false);
+
+if (pause_is_paused)  //draw frozen image to screen while paused
+{
+	surface_set_target(application_surface);
+		if (surface_exists(pause_surface)) draw_surface(pause_surface, 0,0);
+		else //restore from buffer if the surface is lost
+		{
+			pause_surface = surface_create(RES_W,RES_H);
+			buffer_set_surface(pause_surface_buffer,pause_surface,0);
+		}
+	surface_reset_target();
+}
+
+if (keyboard_check_pressed(vk_pause) || keyboard_check_pressed(vk_escape)) //Toggle Pause

```

Comment:
I would move this check into the step (end maybe?) event, and save the first `draw_surface` (at line 28) to the draw event.

Reply:
We can leave this for now.